src/osd/PeeringState.cc

```diff
@@ -2477,7 +2477,7 @@ void PeeringState::proc_master_log(
   }
   update_history(oinfo.history);
   ceph_assert(cct->_conf->osd_find_best_info_ignore_history_les ||
-	 info.last_epoch_started >= info.history.last_epoch_started);
+	      oinfo.last_epoch_started >= info.history.last_epoch_started);
```

Comment:
The following line in the log from https://tracker.ceph.com/issues/41657 indicates that 
info.history.last_epoch_started=42 and oinfo.last_epoch_started=0, even with this condition we'd be hitting the assert, right? or am I missing something?

`
2019-09-04T13:01:52.601+0000 7ff917a01700 10 osd.0 pg_epoch: 49 pg[2.1s1( empty local-lis/les=0/0 n=0 ec=21/21 lis/c=41/23 les/c/f=42/24/0 sis=49) [2147483647,0,4,3]p0(1) r=1 lpr=49 pi=[23,49)/1 crt=0'0 mlcod 0'0 peering mbc={}]  peer osd.3(3) now 2.1s3( DNE empty local-lis/les=0/0 n=0 ec=0/0 lis/c=0/0 les/c/f=0/0/0 sis=0) missing(0 may_include_deletes = 0)
`

Reply:
Yeah, I missed that.